src/app/skills/behavior-detail-alt/behavior-detail-alt.component.ts

```diff
@@ -401,17 +395,40 @@ export class BehaviorDetailAltComponent implements OnInit {
       }
     };
     this.network = new vis.Network(this.container, data, options);
-    this.network.on('select', params => this.select(params));
+    this.network.on('click', params => {
+      this.select(params.nodes[0], "to");
+    });
+    this.network.on('oncontext', params => {
+      params.event.preventDefault();
+      const nodeId = this.network.getNodeAt(params.pointer.DOM);
+      this.select(nodeId, "from");
+    });
   }
 
-  select(params: any): void {
-    if (params.nodes.length > 0) {
-      this.selectedBehavior = this.behaviors[params.nodes[0]];
-      this.selectedBehaviorID = params.nodes[0];
-    } else {
+  select(nodeId, dir): void {
+    if (!nodeId) {
       this.selectedBehavior = undefined;
-      this.selectedBehaviorID = undefined;
+      return;
     }
+    this.router.navigate(['..', nodeId], { relativeTo: this.route });
```

Comment:
Can you make this absolute? This looks brittle to me, e.g. it might break if for some reason the router is on a `/behaviors/:id` url, instead of a `/behaviors/:id/:selected` url

Reply:
With this change, `/behaviors/:id` automatically gets redirected to `/behaviors/:id/:id`. (This is required because otherwise angular reloads the component when the router link changes, and from what I've seen, this is the easiest way to avoid that.)

Reply:
Right, that's done directly in the router. Fair enough I guess. Maybe leave a comment about that if you want.